sensor/common/registry/lazy_tls_image_registry_test.go

```diff
@@ -0,0 +1,283 @@
+package registry
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/buildinfo"
+	imgTypes "github.com/stackrox/rox/pkg/images/types"
+	"github.com/stackrox/rox/pkg/images/utils"
+	"github.com/stackrox/rox/pkg/protoassert"
+	"github.com/stackrox/rox/pkg/registries/docker"
+	"github.com/stackrox/rox/pkg/registries/types"
+	regMocks "github.com/stackrox/rox/pkg/registries/types/mocks"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+)
+
+func TestNoPanics(t *testing.T) {
+	assert.NotPanics(t, func() {
+		failCache := newTLSCheckCache(alwaysFailCheckTLS)
+
+		lazyRegistry := &lazyTLSCheckRegistry{tlsCheckCache: failCache}
+		lazyRegistry.Config()
+		lazyRegistry.DataSource()
+		lazyRegistry.Match(nil)
+		_, _ = lazyRegistry.Metadata(nil)
+		lazyRegistry.Name()
+		lazyRegistry.Source()
+	})
+}
+
+func TestConfig(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockReg := regMocks.NewMockRegistry(ctrl)
+	creator := func(integration *storage.ImageIntegration, options ...types.CreatorOption) (types.Registry, error) {
+		cfg := integration.GetIntegrationConfig().(*storage.ImageIntegration_Docker).Docker
+		mockReg.EXPECT().Config().Return(&types.Config{
+			Insecure: cfg.Insecure,
+		})
+
+		return mockReg, nil
+	}
+
+	source := genImageIntegration("example.com")
+	reg, err := createReg(source, creator, alwaysInsecureCheckTLS)
+	require.NoError(t, err)
+
+	// Before init Insecure should be at the default value (false).
+	assert.False(t, reg.Config().GetInsecure())
+
+	// Simulate some other method triggering lazy init.
+	reg.lazyInit()
+
+	// After init Insecure should come from the backing registry.
+	assert.True(t, reg.Config().GetInsecure())
+}
+
+func TestDataSource(t *testing.T) {
+	ds := &storage.DataSource{}
+	reg := &lazyTLSCheckRegistry{
+		dataSource: ds,
+	}
+
+	got := reg.DataSource()
+	if got != ds {
+		t.Error(t, "Recieved different datasource than expected.")
+	}
```

Comment:
Wanted to compare are the same pointer instead of same contents:

```go
type thing struct {
}

func TestCompare(t *testing.T) {
	t.Run("assert", func(t *testing.T) {
		t1 := &thing{}
		t2 := &thing{}

		assert.Equal(t, t1, t2)
	})

	t.Run("equal", func(t *testing.T) {
		t1 := &thing{}
		t2 := &thing{}

		if t1 != t2 {
			t.Error("NOT equal")
		}
	})
}
```
Output:
```
=== RUN   TestCompare
=== RUN   TestCompare/assert
=== RUN   TestCompare/equal
    /Users/dcaravel/dev/stackrox/stackrox-lazy-tls-check/sensor/common/registry/lazy_tls_image_registry_test.go:301: NOT equal
--- FAIL: TestCompare (0.00s)
    --- PASS: TestCompare/assert (0.00s)
    --- FAIL: TestCompare/equal (0.00s)
FAIL
FAIL	github.com/stackrox/rox/sensor/common/registry	1.692s

```

Reply:
Make sense 👍